### ISSUE TITLE
[Feature] Add config option to include tag & timestamp with File Output plain format

### ISSUE THREAD

**Issue by tinproject** [fluent/fluent-bit#2264]
**Is your feature request related to a problem? Please describe.**

File output `plain format` according to documentation

> Output the records as JSON (without additional tag and timestamp attributes). There is no configuration parameters for plain format.

This will lead to store records in files with incomplete data, as there are no context tag or timestamp.

Add option to store also `tag` and `timestamp` will enable those files to be used to reload the log data to an Elasticsearch server.

**Describe the solution you'd like**

Add options to `File` output like the ones in `Elasticsearch` output so the structure of the file could map the same JSON structure that is sent to Elasticsearch. Config keys:

> Include_Time_Key
Time_Key
Time_Key_Format
Include_Tag_Key
Tag_Key

**Describe alternatives you've considered**

Just add tag & timestamp with fixed names/data format

**Additional context**

**Comment by rayjlinden** [comment]
Why not then instead use json format when does have that data?

**Comment by tinproject** [comment]
@rayjlinden I do not understand. On documentation https://docs.fluentbit.io/manual/pipeline/outputs/file#plain-format states that tag & timestamp are not included on the JSON file output (as quoted above)

My use case is input log lines from a file, parse them, then enrich the data with fluentbit filters to add context. And finally, output records to an Elasticsearch and to a File (for long term archiving).

Basically i just want to store the same data on the File output than the Elasticsearch output. So then I can send in bulk the (ND)JSON file generated by fluentbit to an ES cluster in another point in time (archiving case).

**Comment by rayjlinden** [comment]
Sorry I meant out_file format.

**Comment by tinproject** [comment]
That's what I use right now, but if I want to load the logs on an ES server I have to do transformation from mspack to NDJSON before upload.
This requested feature is to avoid the transformation so we could just send the file in bulk to Elasticsearch.

**Comment by kprasad99** [comment]
Any plan to implement this feature request? I too ended up with same issue. In my case I don't run ES at customer place, at customer place I read data from k8s logs and write to another place. after copying the generated file to my laptop. I push the data to ES using fluentd. This was working until sometime(don't remember the version in which it was broken). Problem here is neither fluentbit nor fluentd provide parser for out_file format.

**Comment by q2dg** [comment]
This issue shouldn't have been closed. See https://github.com/fluent/fluent-bit/issues/2118

**Comment by q2dg** [comment]
I've "solved" this problem adding _-F record="timestamp $(date '+%Y-%m-%dT%T.%NZ')" -m "*"_  to FluentBit's commandline (I can't add this inside config file because bashism "$(...)" isn't interpreted... It's a really ugly hack but it works (or sort of).
